### DISCUSSION TITLE
Parsing responses

### DISCUSSION THREAD

**Discussion by jhurder-cvet** [IlyaMatsuev/Apex-GraphQL-Client#15] [Q&A] [answered]
Hello again! I have made great progress with using your library for making GraphQL queries - now I have a question regarding deserializing the responses. The structure looks like this:

```
query {
  orders {
     edges {
        node {
           id
           name
           ref
           customer { 
              firstName
              lastName
              ref
           }
       }
    }
  } 
}
```

My response looks like this: {orders={edges=({node={ <fields> }. {node={ <fields> }, etc, ...
I can easily get the first node, however is there a technique to parse the rest of the payload as well? Basically, I am looking to get to the information in the `node` element. I have a class that I can deserialize into, but I don't know how to parse any deeper than the 1st element.

Thanks - Jason

**Comment by IlyaMatsuev** [comment, marked as answer]
Hello,
Okay, I see what is the use case. Right now I think there is only one option to do this:
Create the classes structure according to the response:
```java
// These classes can be nested to the class where you actually get the response
// This is just an example, I'm not sure about your structure and what fields are array or object nodes

public class OrderResponse {
    public List<EdgeResponse> edges;
}

public class EdgeResponse {
    public EdgeNodeResponse node;
}

public class EdgeNodeResponse {
    public CustomerResponse customer;
}

public class CustomerResponse {
    public String firstName;
    public String lastName;
    public String ref;
}
```

And then you can do:

```java
...
OrderResponse order = (OrderResponse) response.getDataNodeAs('orders', OrderResponse.class);
System.debug(order.edges[0].customer.firstName);
```

I think this should work if the Apex classes structure is correct

This is of course not the most convenient way to get only one particular field from some deep node for example. But generally, if you have such a complex structure it's always better to use Apex classes instead of just `Map<String, Object>`.

You can also create an issue for introducing a new feature describing why you need it and how you want it to look (if you don't like the way described above)

**Reply by jhurder-cvet** [reply]
Awesome - thank you for the direction, this is working well!
